server/store/flow_store.go

```diff
@@ -6,26 +6,28 @@ const (
 	UpdateStatusPropertyName              = "update_status"
 	GetConfirmationPropertyName           = "get_confirmation"
 	ReceiveNotificationsDuringMeetingName = "receive_notifications_during_meetings"
+	SetCustomStatusPropertyName           = "set_custom_status"
 	SubscribePropertyName                 = "subscribe"
 	ReceiveUpcomingEventReminderName      = "receive_reminder"
 )
 
-func (s *pluginStore) SetProperty(userID, propertyName string, value bool) error {
+func (s *pluginStore) SetProperty(userID, propertyName string, value interface{}) error {
 	user, err := s.LoadUser(userID)
 	if err != nil {
 		return err
 	}
 
+	boolValue := value.(bool)
 	switch propertyName {
 	case UpdateStatusPropertyName:
-		user.Settings.UpdateStatus = value
-		s.Tracker.TrackAutomaticStatusUpdate(userID, value, "flow")
+		user.Settings.UpdateStatusFromOptions = value.(string)
+		s.Tracker.TrackAutomaticStatusUpdate(userID, value.(string), "flow")
```

Comment:
Type assertions with a check variable do not panic if the type doesn't match.

> The value of ok is true if the assertion holds. Otherwise it is false and the value of v is the [zero value](https://go.dev/ref/spec#The_zero_value) for type T. No [run-time panic](https://go.dev/ref/spec#Run_time_panics) occurs in this case.

 https://go.dev/ref/spec#Type_assertions

See https://goplay.tools/snippet/ETzO-4sgImZ as an example. Only L8 panics, not L6.

Reply:
@hanzei Yes but there was no check variable on this before